### DISCUSSION TITLE
Cert expiration in helm chart for auto generated sp-validator, proxy-injector and tap

### DISCUSSION THREAD

**Discussion by twendt** [linkerd/linkerd2#4977] [General] [answered]
The helm chart uses a fixed cert expiration of 365 days for the proxy injector, sp-validator and tap. Does that mean that the cert has to be manually renewed after 365 days?

Here is the link to the code in the sp-validator helm template:
https://github.com/linkerd/linkerd2/blob/599872815884f6b5dd5eddcad0af4a57b4248ad7/charts/linkerd2/templates/sp-validator-rbac.yaml#L45

**Comment by ihcsim** [comment, marked as answer]
Yes, the `linkerd upgrade` and `helm upgrade` commands can be used to renew those certs. See https://linkerd.io/2/tasks/rotating_webhooks_certificates/. Also, some folks in the community have been trying to put together steps on how to automate this with cert-manager.

**Comment by twendt** [comment]
Great, thank you. That was exactly the hint I was looking for.
